fe/fe-core/src/main/antlr4/org/apache/doris/nereids/PLParser.g4

```diff
@@ -324,11 +325,15 @@ package_body_item :
       declare_stmt_item
     | create_function_stmt
     | create_procedure_stmt
+    | drop_procedure_stmt
     ;
 
 create_procedure_stmt :
       (ALTER | CREATE (OR REPLACE)? | REPLACE) (PROCEDURE | PROC) multipartIdentifier create_routine_params? create_routine_options? (AS | IS)? declare_block_inplace? label_stmt? procedure_block (ident_pl SEMICOLON)?
     ;
+drop_procedure_stmt:
+      (DROP  (PROCEDURE | PROC) multipartIdentifier create_routine_params? create_routine_options?)
```

Comment:
not need `create_routine_params` and `create_routine_options`, refer to /hive/hplsql/src/main/antlr4/org/apache/hive/hplsql/Hplsql.g4, our syntax is the same as hplsql
![image](https://github.com/apache/doris/assets/13197424/16e17046-caff-4654-a6a6-2f34d748142f)

such as:
`DROP PROCEDURE (IF EXISTS)? name=multipartIdentifier`
or
`DROP (PROCEDURE | PROC) (IF EXISTS)? name=multipartIdentifier`